orbit/cmd/desktop/desktop_windows.go

```diff
@@ -99,3 +100,52 @@ func watchSystemTheme(iconManager *iconManager) {
 		}()
 	}
 }
+
+// blockWaitForStopEvent waits for the named event kernel object to be signalled
+func blockWaitForStopEvent(channelId string) error {
+	if channelId == "" {
+		return errors.New("communication channel name should not be empty")
+	}
+
+	// converting go string to UTF16 windows compatible string
+	targetChannel := "Global\\comm-" + channelId
+	ev, err := windows.UTF16PtrFromString(targetChannel)
+	if err != nil {
+		return fmt.Errorf("there was a problem generating UTF16 string: %w", err)
+	}
+
+	log.Info().Msg("Comm channel was acquired ")
+
+	// The right to use the object for synchronization
+	// https://learn.microsoft.com/en-us/windows/win32/sync/synchronization-object-security-and-access-rights
+	const EVENT_SYNCHRONIZE = 0x00100000
+
+	// block wait until channel is available is available
+	var handle windows.Handle = windows.InvalidHandle
+	for {
+		// OpenEvent API opens a named event object from the kernel object manager
+		// https://learn.microsoft.com/en-us/windows/win32/api/synchapi/nf-synchapi-openeventw
+		handle, err = windows.OpenEvent(EVENT_SYNCHRONIZE, false, ev)
```

Comment:
Should the error be logged? (I have no idea how often or why `windows.OpenEvent` can fail.)

Reply:
this call is done inside a loop, so I didn't want to log on purpose as the call might fail initially until the named event kernel object is available. The idea here is that blockWaitForStopEvent() will execute on its own runner to pool the named object until it gets available. It will check every x amount of ms if the named event object is there and continue only when handle is acquired basically. The reason for this is to ensure fleet-desktop is capable of listening to signaling events on this object, independently of the object creation or signaling time. The event creation happens during orbit startup, and the signaling time can happen multiple times of the orbit lifecycle (every time that SignalProcessBeforeTerminate() is basically called). So having an asynchronous pooling logic like this ensures no race conditions between fleet-desktop and orbit happen.